geomfn: fix st_simplify with NaN

techytoes opened the pull request:
Release note (SQL Change): Earlier st_simplify with NaN caused node to crash.
This fixes the behaviour to align it more to PostGIS.

Fixes: #63620

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63760)
<!-- Reviewable:end -->

otan commented on pkg/geo/geomfn/topology_operations.go:
```diff
@@ -96,6 +98,9 @@ func Difference(a, b geo.Geometry) (geo.Geometry, error) {
 
 // Simplify returns a simplified Geometry.
 func Simplify(g geo.Geometry, tolerance float64) (geo.Geometry, error) {
+	if math.IsNaN(tolerance) {
```
can you change this to `if math.IsNaN(tolerance) g.ShapeType2D() == geopb.ShapeType_POINT || g.ShapeType2D() == geopb.ShapeType_MULTIPOINT `, and add tests for points and multipoints?

techytoes replied:
I have made the changes. please review this again.

Thanks🙇🏻‍♂️

Commit:
geomfn: fix st_simplify with NaN

Release note (SQL Change): Earlier st_simplify with NaN caused node to crash.
This fixes the behaviour to align it more to PostGIS.

otan reviewed (approved):
looks good! thanks!

bors r+

techytoes commented:
TFTR! @otan 🙇🏻‍♂️
